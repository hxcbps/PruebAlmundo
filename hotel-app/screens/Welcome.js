import React from 'react';
import { Text, View, Image, Dimensions } from 'react-native';
import { Button } from 'react-native-elements';
import { Spinner } from '../components/commons';
import { appMainColor } from '../const';

const SCREEN_WIDTH = Dimensions.get('window').width - 30;

export default class App extends React.Component {
  state = { loading: false };

  onButtonPress() {
    this.props.navigation.navigate('HotelListScreen');
  }

  renderLoginButton() {
    if (this.state.loading) {
      return <Spinner size="large" />;
    }

    return (
      <Button
        raised
        borderRadius={5}
        backgroundColor={appMainColor}
        buttonStyle={{ alignSelf: 'stretch' }}
        icon={{ name: 'sign-in', type: 'font-awesome' }}
        title='Ingresar'
        onPress={this.onButtonPress.bind(this)}
      />
    );
  }

  render() {
    return (
      <View style={styles.logoContainer}>
        <Image
          resizeMode="contain"
          source={require('../assets/images/logo-letras.png')}
          style={styles.mainLogo}
        />
        <Text
          style={styles.welcomeText}
        >
          Bienvenido, con nuestra app podr√° consultar todos nuestros hoteles a nivel mundial
        </Text>
        {this.renderLoginButton()}
      </View>
    );
  }
}

const styles = {
  logoContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#fff'
  },
  mainLogo: {
    width: SCREEN_WIDTH,
    justifyContent: 'center',
    margin: 10
  },
  welcomeText: {
    alignSelf: 'center',
    fontSize: 20,
    padding: 20
  }
};
