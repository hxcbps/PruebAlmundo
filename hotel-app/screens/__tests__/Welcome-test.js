import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Welcome from '../Welcome';


test('renders correctly', () => {
  const tree = renderer.create(
    <Welcome />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
