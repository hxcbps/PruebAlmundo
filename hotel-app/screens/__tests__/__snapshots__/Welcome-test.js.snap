// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "justifyContent": "center",
        "margin": 10,
        "width": 720,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 20,
        "padding": 20,
      }
    }
  >
    Bienvenido, con nuestra app podrá consultar todos nuestros hoteles a nivel mundial
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#ff7900",
            },
            Object {
              "borderRadius": 5,
            },
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "alignSelf": "stretch",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="font-awesome"
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Ingresar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
